frontend/src/routes/_oh.app.tsx

```diff
@@ -156,21 +157,32 @@ function App() {
     );
   };
 
+  const doSendInitialQuery = React.useRef<boolean>(true);
```

Comment:
Although there us nothing wrong with this approach, it may be preferable to convert this into a `useState` since you change the value in `handleOpen`.

One key difference though is that a `useRef` **does not** cause a re-render whereas setting a `useState` does. If that is what you intend then don't worry about it.

Reply:
I suppose a refactor is in order, we probably don't need to wrap all functions in callbacks. Let me know on how you plan on extending the WS logic when you come to it later on, I had some ideas too (with multi session coming up)